test/shift_generator_test.rb

```diff
@@ -12,44 +12,30 @@ def test_it_exists
   end
 
   def test_it_has_attributes
+    assert_instance_of KeyGen, @shift_generator.key
+    assert_instance_of OffsetGen, @shift_generator.offset
     @shift_generator.stubs(:generate_random_key).returns("02715")
-    assert_instance_of String, @shift_generator.key
-    assert_equal 5, @shift_generator.key.length
-    assert_equal "260220", @shift_generator.date
+    assert_instance_of String, @shift_generator.key.num
+    assert_equal 5, @shift_generator.key.num.length
+    assert_equal "260220", @shift_generator.offset.date
   end
 
   def test_it_can_generate_random_five_digit_number_key
-    assert_instance_of String, @shift_generator.generate_random_key
-    assert_equal 5, @shift_generator.generate_random_key.to_s.length
+    assert_instance_of String, @shift_generator.key.generate_random_key
+    assert_equal 5, @shift_generator.key.generate_random_key.to_s.length
   end
 
   def test_it_can_split_return_value_of_generate_random_key_into_separate_keys
     expected = { :key_A => 02, :key_B => 27, :key_C => 71, :key_D => 15}
-    assert_equal expected, @shift_generator.separate_random_key("02715")
+    assert_equal expected, @shift_generator.key.create_key("02715")
 
-    random_test_key = @shift_generator.generate_random_key
-    assert_instance_of Hash, @shift_generator.separate_random_key(random_test_key)
-    assert_equal 4, @shift_generator.separate_random_key(random_test_key).values.length
-  end
-
-  def test_it_can_square_date_parameter
-    test_date = "270220"
-    assert_equal "73018848400", @shift_generator.square_date(test_date)
-  end
-
-  def test_it_can_get_last_4_digits_from_sqaure_date
-    squared_date_param = "73018848400"
-    assert_equal "8400", @shift_generator.last_four_digits(squared_date_param)
-  end
-
-  def test_it_can_separate_last_4_digits_from_square_dates_into_keys
-    four_digits = "8400"
-    expected = {key_A: 8, key_B: 4, key_C: 0, key_D: 0}
-    assert_equal expected, @shift_generator.separate_last_4_digits(four_digits)
+    random_test_key = @shift_generator.key.generate_random_key
+    assert_instance_of Hash, @shift_generator.key.create_key(random_test_key)
+    assert_equal 4, @shift_generator.key.create_key(random_test_key).values.length
   end
 
   def test_it_can_get_final_shift_values
     expected = {key_A: 10, key_B: 31, key_C: 71, key_D: 15}
-    assert_equal expected, @shift_generator.final_shift_values("02715", "8400")
+    assert_equal expected, @shift_generator.shift_key("02715", "8400")
   end
 end
```

Comment:
had to test key and offset in shift class to make sure integration is working, coverage is at 100% at the moment